primitives/traits/src/lib.rs

```diff
@@ -31,23 +31,35 @@ use {
     sp_std::vec::Vec,
 };
 
-/// The collator-assignment hook to react to collators beign assigned to container chains.
-pub trait CollatorAssignmentHook {
+/// The collator-assignment hook to react to collators being assigned to container chains.
+pub trait CollatorAssignmentHook<Balance> {
     /// This hook is called when collators are assigned to a container
     ///
     /// The hook should never panic and is required to return the weight consumed.
-    fn on_collators_assigned(para_id: ParaId) -> Weight;
+    fn on_collators_assigned(para_id: ParaId, maybe_tip: &Option<Balance>) -> Weight;
```

Comment:
```suggestion
    fn on_collators_assigned(para_id: ParaId, maybe_tip: Option<&Balance>) -> Weight;
```

This is always better, use `.as_ref()` to convert from `&Option` to `Option<&`